### ISSUE TITLE
Refine for real/imaginary part

### ISSUE THREAD

**Issue by hoba87** [sympy/sympy#17011]
The following code:
a=symbols("a", positive=True)
b=symbols("b")
print(re(a))
print(refine(re(b), Q.positive(b)))

give as output:
a
re(b)

Is that the correct behaviour? I would have expected as output:
a
b

**Comment by vighneshq** [comment]
```
handlers_dict = {
    'Abs': refine_abs,
    'Pow': refine_Pow,
    'atan2': refine_atan2,
    'Equality': refine_Relational,
    'Unequality': refine_Relational,
    'GreaterThan': refine_Relational,
    'LessThan': refine_Relational,
    'StrictGreaterThan': refine_Relational,
    'StrictLessThan': refine_Relational
}
```
These are the only functions for which refine would work. There is no handler  currently defined for ``` re ```.

**Comment by vighneshq** [comment]
If it is okay, I can add the handlers for ``` re ```, ``` im ``` methods.

**Comment by oscarbenjamin** [comment]
That would be great. Thanks.

**Comment by oscarbenjamin** [comment]
This is fixed now by #17019:
```julia
In [8]: a=symbols("a", positive=True) 
   ...: b=symbols("b") 
   ...: print(re(a)) 
   ...: print(refine(re(b), Q.positive(b)))                                                                                       
a
b
```
